DuckDuckGo/VPNWaitlist.swift

```diff
@@ -0,0 +1,126 @@
+//  limitations under the License.
+//
+
+#if NETWORK_PROTECTION
+
+import Foundation
+import BrowserServicesKit
+import Combine
+import Core
+import Waitlist
+import NetworkProtection
+
+final class VPNWaitlist: Waitlist {
+
+    enum AccessType {
+        /// Used if the user does not have waitlist feature flag access
+        case none
+
+        /// Used if the user has waitlist feature flag access, but has not joined the waitlist
+        case waitlistAvailable
+
+        /// Used if the user has waitlist feature flag access, and has joined the waitlist
+        case waitlistJoined
+
+        /// Used if the user has been invited via the waitlist, but needs to accept the Privacy Policy and Terms of Service
+        case waitlistInvitedPendingTermsAcceptance
+
+        /// Used if the user has been invited via the waitlist and has accepted the Privacy Policy and Terms of Service
+        case waitlistInvited
+
+        /// Used if the user has been invited to test Network Protection directly
+        case inviteCodeInvited
+    }
+
+    static let identifier: String = "vpn"
+    static let apiProductName: String = "networkprotection_ios"
+    static let downloadURL: URL = URL.windows
+
+    static let shared: VPNWaitlist = .init()
+
+    static let backgroundTaskName = "VPN Waitlist Status Task"
+    static let backgroundRefreshTaskIdentifier = "com.duckduckgo.app.vpnWaitlistStatus"
+    static let notificationIdentifier = "com.duckduckgo.ios.vpn.invite-code-available"
+    static let inviteAvailableNotificationTitle = UserText.networkProtectionWaitlistNotificationTitle
+    static let inviteAvailableNotificationBody = UserText.networkProtectionWaitlistNotificationText
+
+    var isAvailable: Bool {
+        isFeatureEnabled
+    }
+
+    var isWaitlistRemoved: Bool {
+        return false
+    }
+
+    let waitlistStorage: WaitlistStorage
+    let waitlistRequest: WaitlistRequest
+    private let privacyConfigurationManager: PrivacyConfigurationManaging
+    private let networkProtectionAccess: NetworkProtectionAccess
+
+    init(store: WaitlistStorage,
+         request: WaitlistRequest,
+         privacyConfigurationManager: PrivacyConfigurationManaging = ContentBlocking.shared.privacyConfigurationManager,
+         networkProtectionAccess: NetworkProtectionAccess = NetworkProtectionAccessController()) {
+        self.waitlistStorage = store
+        self.waitlistRequest = request
+        self.privacyConfigurationManager = privacyConfigurationManager
+        self.networkProtectionAccess = networkProtectionAccess
+
+        let hasWaitlistAccess = privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlist)
+        let isWaitlistActive = privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlistBetaActive)
+        let internalUser = InternalUserStore()
+        isFeatureEnabled = (hasWaitlistAccess && isWaitlistActive) || internalUser.isInternalUser
```

Comment:
I added an internal user check here, but I'm convinced that we really need it. These flags are enabled remotely now, but the current copy of the app isn't bundled with a copy that has that change, hence the internal check.

Reply:
I’m not sure I understand this. I don’t have to block the PR, but I think the FeatureFlagger may cover what you need here.

Reply:
Yeah I don't think we need it either, let me look at updating this.